utils/build-script-impl

```diff
@@ -3158,16 +3185,11 @@ for host in "${ALL_HOSTS[@]}"; do
 
             # Run dsymutil on executables and shared libraries.
             #
-            # Exclude shell scripts and static archives.
-            # Exclude swift-api-digester dSYM to reduce debug toolchain size.
             # Tweak carefully the amount of parallelism -- dsymutil can be memory intensive and
             # as such too many instance can exhaust the memory and slow down/panic the machine
             printJSONStartTimestamp dsymutil
             (cd "${host_symroot}" &&
              find ./"${CURRENT_PREFIX}" -perm -0111 -type f -print | \
-               grep -v '.py$' | \
-               grep -v '.a$' | \
```

Comment:
This part of the change (moving the grep -v of the .py/.a/etc) seems to not be about generate symbols for a subset of binaries, can you do a git reset --soft and split this into a separate commit (same PR) that has a commit message that explains why this is being done. I am fine if you leave the test in the 2nd commit and just say in the first commit msg that you are putting the test in the 2nd commit.

Reply:
After considering also the feedback in https://github.com/apple/swift/pull/37120#discussion_r626077819, I broke the PR in two commits as follows:

* fc5fa646715af4862655da49d87fcfb7fe6b36b3 focuses on getting this section under test and capturing the current behaviour (especially around populating the symroot when generating the toolchains)
* d3fb1c78b24f2c913dd59c0774fa1dfe1835608a introduce the filtering logic to copy only some paths to the symroot.